thor/tests/e2e.rs

```diff
@@ -575,6 +566,198 @@ async fn e2e_splice_in() {
+    let bob_update = bob_channel.update_balance(
+        &mut bob_transport,
+        Balance {
+            ours: expected_bob_balance,
+            theirs: expected_alice_balance,
+        },
+        time_lock,
+    );
+
+    futures::future::try_join(alice_update, bob_update)
+        .await
+        .unwrap();
+
+    // Assert: Channel balances are correct
+
+    assert_eq!(expected_alice_balance, alice_channel.balance().ours);
+    assert_eq!(expected_bob_balance, bob_channel.balance().ours);
+
+    assert_eq!(alice_channel.balance().ours, bob_channel.balance().theirs);
+    assert_eq!(alice_channel.balance().theirs, bob_channel.balance().ours);
+
+    let Balance {
+        ours: actual_alice_balance,
+        theirs: actual_bob_balance,
+    } = alice_channel.balance();
+
+    // Act: Splice-out the channel:
+    //  Bob withdraws 1.2BTC, decreasing his channel balance to 0.1 BTC
+    let bob_splice_address = bob_wallet.0.new_address().await.unwrap();
+    let bob_splice_out_amount = Amount::from_btc(1.2).unwrap();
+    let bob_splice = Splice::Out(TxOut {
+        script_pubkey: bob_splice_address.script_pubkey(),
+        value: bob_splice_out_amount.as_sat(),
+    });
+
+    let before_splice_bob_balance = bob_wallet.0.balance().await.unwrap();
+
+    let alice_splice = alice_channel.splice(&mut alice_transport, &alice_wallet, Splice::None);
+    let bob_splice = bob_channel.splice(&mut bob_transport, &bob_wallet, bob_splice);
+
+    let (mut alice_channel, mut bob_channel) = futures::future::try_join(alice_splice, bob_splice)
+        .await
+        .unwrap();
+
+    // Assert: Channel balances are correct
+
+    let splice_out_fee = Amount::from_sat(thor::TX_FEE);
+
+    assert_eq!(actual_alice_balance, alice_channel.balance().ours);
+    assert_eq!(
+        actual_bob_balance - bob_splice_out_amount - splice_out_fee,
+        bob_channel.balance().ours
+    );
+
+    assert_eq!(alice_channel.balance().ours, bob_channel.balance().theirs);
+    assert_eq!(alice_channel.balance().theirs, bob_channel.balance().ours);
+
+    let Balance {
+        ours: actual_alice_balance,
+        theirs: actual_bob_balance,
+    } = alice_channel.balance();
+
+    bitcoind_fee_bob +=
+        before_splice_bob_balance + bob_splice_out_amount - bob_wallet.0.balance().await.unwrap();
+
+    // Act: Alice pays Bob 0.5 BTC
+
+    let payment = Amount::from_btc(0.5).unwrap();
+    let expected_alice_balance = actual_alice_balance - payment;
+    let expected_bob_balance = actual_bob_balance + payment;
+
+    let alice_update = alice_channel.update_balance(
+        &mut alice_transport,
+        Balance {
+            ours: expected_alice_balance,
+            theirs: expected_bob_balance,
+        },
+        time_lock,
+    );
+    let bob_update = bob_channel.update_balance(
+        &mut bob_transport,
+        Balance {
+            ours: expected_bob_balance,
+            theirs: expected_alice_balance,
+        },
+        time_lock,
+    );
```

Comment:
All this stuff is going to cause merge conflict with the e2e cleanup [I did](https://github.com/comit-network/thor/pull/57). If you want I can fix the e2e stuff in this PR inline with the other changes, if they happen to merge before hand.